### ISSUE TITLE
clangd.inlayHints.enable=false seems not working when I upgrade coc-clangd to 0.24

### ISSUE THREAD

**Issue by Layamon** [clangd/coc-clangd#499]
Please describe the problem.
For hints on what information is helpful, see: https://clangd.llvm.org/troubleshooting.html

If you can, provide a minimal chunk of code that shows the problem (either inline, or attach it if larger).

**Logs**
Please attach the clangd log from `:CocCommand workspace.showOutput` if you can.
If possible, run with `--log=verbose` - note that the logs will include the contents of open files!

**System information**
Clangd version (from the log, or `clangd --version`):
clangd extension version (from `:CocInfo`):
Operating system:
  33
  32 vim version: NVIM v0.7.0
  31 node version: v14.19.3
  30 coc.nvim version: 0.0.82-3ab29075 2022-08-26 00:21:58 +0800
  29 coc.nvim directory: /home/lym/.local/share/nvim/plugged/coc.nvim
  28 term: tmux
  27 platform: linux
  26
  25 ## Log of coc.nvim
  24
  23 2022-09-06T10:22:36.992 ERROR (pid:5815) [configurations] - Invalid configuration "suggest.detailMaxLength" from coc-clangd, overwrite defaults is fobidden.
  22 2022-09-06T10:22:36.993 ERROR (pid:5815) [configurations] - Invalid configuration "diagnostic.format" from coc-clangd, overwrite defaults is fobidden.
  21 2022-09-06T10:22:37.211 INFO (pid:5815) [services] - registered service "json"
  20 2022-09-06T10:22:37.212 INFO (pid:5815) [services] - Json language server state change: stopped => starting
  19 2022-09-06T10:22:37.216 INFO (pid:5815) [plugin] - coc.nvim initialized with node: v14.19.3 after 252ms
  18 2022-09-06T10:22:37.259 INFO (pid:5815) [services] - registered service "clangd"
  17 2022-09-06T10:22:37.259 INFO (pid:5815) [services] - clangd state change: stopped => starting
  16 2022-09-06T10:22:37.291 INFO (pid:5815) [language-client-index] - Language server "clangd" started with 5835
  15 2022-09-06T10:22:37.307 INFO (pid:5815) [services] - clangd state change: starting => running
  14 2022-09-06T10:22:37.316 INFO (pid:5815) [services] - service clangd started
  13 2022-09-06T10:22:37.371 INFO (pid:5815) [services] - Json language server state change: starting => running
  12 2022-09-06T10:22:37.380 INFO (pid:5815) [services] - service json started
  11 2022-09-06T10:22:40.851 INFO (pid:5815) [attach] - receive notification: highlight []
  10 2022-09-06T10:22:54.841 INFO (pid:5815) [attach] - receive notification: highlight []
   9 2022-09-06T10:24:23.894 INFO (pid:5815) [attach] - receive notification: highlight []
   8 2022-09-06T10:24:25.102 INFO (pid:5815) [attach] - receive notification: runCommand [ 'workspace.showOutput' ]
   7 2022-09-06T10:24:30.458 WARN (pid:5815) [plugin] - Slow action "runCommand" cost 5352ms
   6 2022-09-06T10:24:30.938 INFO (pid:5815) [attach] - receive notification: highlight []
   5 2022-09-06T10:24:32.494 INFO (pid:5815) [attach] - receive notification: highlight []
   4 2022-09-06T10:24:34.150 INFO (pid:5815) [attach] - receive notification: highlight []
   3 2022-09-06T10:24:35.088 INFO (pid:5815) [attach] - receive notification: highlight []
   2 2022-09-06T10:24:36.097 INFO (pid:5815) [attach] - receive notification: highlight []
   1 2022-09-06T10:24:42.795 INFO (pid:5815) [attach] - receive notification: showInfo []
 35

**Comment by fannheyward** [comment]
What do you mean `seems not working`? `"clangd.inlayHints.enable": false` still show inlay hints?

**Comment by Layamon** [comment]
> What do you mean `seems not working`? `"clangd.inlayHints.enable": false` still show inlay hints?

yes,
this is my cocconfig
```json
{
  "clangd.path": "~/.local/bin/clangd",
  "clangd.arguments": [
    "-cross-file-rename",
    "--fallback-style=Google",
    "--all-scopes-completion",
    "--function-arg-placeholders",
    "--inlay-hints=false",
    "--header-insertion=iwyu"
  ],
  "clangd.inlayHints.enable": false,
  "clangd.inlayHints.sep": "‣",
  "suggest.completionItemKindLabels": {
        "keyword": "\uf1de",
        "variable": "\ue79b",
        "value": "\uf89f",
        "operator": "\u03a8",
        "constructor": "\uf0ad",
        "function": "\u0192",
        "reference": "\ufa46",
        "constant": "\uf8fe",
        "method": "\uf09a",
        "struct": "\ufb44",
        "class": "\uf0e8",
        "interface": "\uf417",
        "text": "\ue612",
        "enum": "\uf435",
        "enumMember": "\uf02b",
        "module": "\uf40d",
        "color": "\ue22b",
        "property": "\ue624",
        "field": "\uf9be",
        "unit": "\uf475",
        "event": "\ufacd",
        "file": "\uf723",
        "folder": "\uf114",
        "snippet": "\ue60b",
        "typeParameter": "\uf728",
        "default": "\uf29c"
    }
}
```
and 
<img width="539" alt="image" src="https://user-images.githubusercontent.com/4601177/188537563-18bfcc28-828d-457a-b514-2dfa8ace110b.png">

**Comment by fannheyward** [comment]
Your configurations are correct. What's your clangd version? Maybe there is something wrong, needs more info to debug.

**Comment by Layamon** [comment]
```
$ ~/.local/bin/clangd --version
clangd version 15.0.0 (/home/lym/archive/llvm-project/clang a67beef3acac4ab4fc009f492d5b3f510c48420b)
Features: linux
```

**Comment by fannheyward** [comment]
Reproduced. Works on clangd 14, but not 15 and 16 nightly.

**Comment by Layamon** [comment]
thanks, I try to downgrade to clangd 14

**Comment by fannheyward** [comment]
Fixed in 0.25.0

**Comment by bear24rw** [comment]
Incase someone else comes here from google the key was removed: https://github.com/clangd/coc-clangd/commit/b1d4e166bc742cf826bd54b53e22e7c193a1d299
